### ISSUE TITLE
`train.py` lacks deterministic seed handling

### ISSUE THREAD

**Issue by gelluisaac** [Traqora/astroml#189]
Description:
Training scripts don't expose seeds consistently causing non-reproducible results.

Suggested fix:
- Add `--seed` flag and set seeds for numpy, random, torch.

---

**Comment by extolkom** [comment]
@extolkom has applied to work on this issue as part of the **Stellar Wave Program**'s 5th wave.



> please assign me


ℹ️ **Repo Maintainers**: To accept this application, [review their application](https://www.drips.network/wave/maintainers/issues/6dd679a0-ae23-479b-8049-ef9eb160eae9) or assign @extolkom to this issue.

**Comment by Themancalledpg** [comment]
@Themancalledpg has applied to work on this issue as part of the **Stellar Wave Program**'s 5th wave.



> can i work on this issue?


ℹ️ **Repo Maintainers**: To accept this application, [review their application](https://www.drips.network/wave/maintainers/issues/6dd679a0-ae23-479b-8049-ef9eb160eae9) or assign @Themancalledpg to this issue.
